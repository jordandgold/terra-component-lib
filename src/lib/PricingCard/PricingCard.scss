@import "../global-styles/terra.scss";

.ter-pricing-card {
  @include box-shadow(2);
  padding: spacing("sm");

  &__product {
    font-weight: font-weight("light");
    text-align: center;
    margin-bottom: 0;
  }

  &__version {
    font-weight: font-weight("extra-bold");
    text-transform: uppercase;
    text-align: center;
    margin-top: 0;
  }

  &__description {
    font-weight: font-weight("light");
    text-align: center;
    font-size: font-size("sm");
    margin-bottom: spacing("xs");
  }

  &__cost {
    font-weight: font-weight("extra-bold");
    text-align: center;
    font-size: font-size("xl");
    margin-bottom: 0;
  }

  &__currency {
    font-weight: font-weight("light");
    text-align: center;
    font-size: font-size("micro");
  }

  .ter-button {
    margin: 0 auto;
    display: block;
  }
}
