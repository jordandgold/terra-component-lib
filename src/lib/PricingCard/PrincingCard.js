import React from "react";
import PropTypes from "prop-types";
import "./PricingCard.scss";
import ButtonLink from "../ButtonLink/ButtonLink";

const PricingCard = ({
  product,
  version,
  description,
  cost,
  currency,
  buttonLink
}) => {
  return (
    <article className="ter-pricing-card">
      <p className="ter-pricing-card__product">{product}</p>
      <p className="ter-pricing-card__version">{version}</p>
      <p className="ter-pricing-card__description">{description}</p>
      <h3 className="ter-pricing-card__cost">{cost}</h3>
      <p className="ter-pricing-card__currency">{currency}</p>
      <ButtonLink
        link={buttonLink.link}
        text={buttonLink.text}
        className="ter-button--secondary--1"
      />
    </article>
  );
};

export default PricingCard;

PricingCard.propTypes = {
  product: PropTypes.string.isRequired,
  version: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  cost: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  buttonLink: PropTypes.shape({
    link: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired
};
