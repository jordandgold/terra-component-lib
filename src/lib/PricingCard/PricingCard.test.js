import React from "react";
import { shallow } from "enzyme";
import PricingCard from "./PrincingCard";

describe("PricingCard", () => {
  let wrapper;

  beforeEach(() => {
    const mockDescription =
      "Ullamco exercitation excepteur nostrud ipsum veniam est enim.";

    const mockButtonLink = {
      link: "#",
      text: "Subscribe"
    };

    wrapper = shallow(
      <PricingCard
        product="SketchUp"
        version="Shop"
        description={mockDescription}
        cost="$119/yr"
        currency="USD"
        buttonLink={mockButtonLink}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
