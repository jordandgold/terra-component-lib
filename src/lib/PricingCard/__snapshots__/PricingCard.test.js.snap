// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard should match the snapshot 1`] = `
<article
  className="ter-pricing-card"
>
  <p
    className="ter-pricing-card__product"
  >
    SketchUp
  </p>
  <p
    className="ter-pricing-card__version"
  >
    Shop
  </p>
  <p
    className="ter-pricing-card__description"
  >
    Ullamco exercitation excepteur nostrud ipsum veniam est enim.
  </p>
  <h3
    className="ter-pricing-card__cost"
  >
    $119/yr
  </h3>
  <p
    className="ter-pricing-card__currency"
  >
    USD
  </p>
  <ButtonLink
    className="ter-button--secondary--1"
    link="#"
    text="Subscribe"
  />
</article>
`;
